File: app/schemas/comment.py
from pydantic import BaseModel
from datetime import datetime
class CommentCreate(BaseModel):
  article_id:int
  content:str
  parent_id:int|None=None

class CommentStatusUpdate(BaseModel):

    status:str

class CommentFrontResponse(BaseModel):

    id:int

    content:str

    user_id:int

    article_id:int

    parent_id:int | None=None

    create_time:datetime

    status:str

    likes:int


    class Config:
        from_attributes=True

class CommentResponse(BaseModel):

    id:int

    article_id:int

    user_id:int

    content:str

    parent_id:int|None=None

    status:str

    likes:int

    create_time:datetime

    article_title:str|None=None

    username:str|None=None


    class Config:
        from_attributes=True

File: src/types/comment.ts
export interface CommentType{
  id:number;
  articleId:number;
  articleTitle?:string;
  userId:number
  username?:string;
  content:string;
  status:string;
  time:string;
  parentId:number|null;
  like:number;
  liked?:boolean;
  parentName?:string
}

// 后端返回的数据
export interface BackendCommentType {
  id: number
  article_id: number
  user_id: number
  article_title:string
  content: string
  parent_id: number | null
  status: string
  create_time: string
}

export interface CommentPage{

 list:BackendCommentType[]

 total:number

}
